### PR TITLE
Limit failing listener to a configured MaxAttempts

### LINKED ISSUE #1164: How can you limit retries on exception when using RabbitListener?
Unable to Limit Retries Using Steeltoe with RabbitMQ in ASP.NET Core 6.0 Microservices

 - .NET Version [NET Core 6.0] 
 - Steeltoe Version [3.1.3]

I'm working on an ASP.NET Core microservices application where I'm utilizing the Steeltoe package to integrate RabbitMQ for messaging. I'm trying to implement a retry mechanism with a maximum number of attempts using the MaxAttempts property provided by the Steeltoe package, but I'm facing difficulties in achieving the desired behavior. Here's the relevant part of my appsettings.json configuration:

`"Spring": {
  "RabbitMq": {
    "Addresses": "*",
    "Template": {
      "Retry": {
        "Enabled": true,
        "MaxAttempts": 3
      }
    }
  }
}`

These are the steeltoe libraries that I am using:
`using Steeltoe.Messaging.RabbitMQ.Attributes;
using Steeltoe.Messaging.RabbitMQ.Core;`

I've also configured RabbitMQ services in my Program.cs as follows:
`builder.Services.AddRabbitServices(true);
builder.Services.AddRabbitAdmin();
builder.Services.AddRabbitTemplate();`

In my application, I have a RabbitMQ message listener method that processes messages. I want to restrict the number of retries for this listener using the MaxAttempts configuration, but despite setting the value to 3 in the configuration, the listener continues to retry indefinitely.

Here's a simplified version of my message listener method:
`[RabbitListener(Constants.RECEIVE_AND_CONVERT_QUEUE)]
public void ListenForAMessage(ContentMessage msg)
{
    // Message processing
    // ...
}`

For better assistance, I've created a minimal example project on GitHub to showcase the problem. In this repository, you can find the simplified version of my application code. I'd greatly appreciate it if you could take a look at the project and provide insights on why the retry mechanism isn't working as expected. 

GitHub Project:  https://github.com/Bassam-Rahat/SteelToe-RabbitMq-Application

Thank you in advance for your help and expertise.

### PR DESCRIPTION
**Pull request by hananiel**
## Description

Currently the RabbitMQ container configured by the `RabbitListener` does not respect the configured `MaxAttempts` value. Adding the required PollyRetry and callback as requested by the configuration. 

Fixes #1164 

## Quality checklist

- [x] Your code complies with our [Coding Style](https://github.com/SteeltoeOSS/.github/blob/main/contributing-docs/contributing-code-style.md).
- [x] You've updated unit and/or integration tests for your change, where applicable.
- [ ] You've updated documentation for your change, where applicable.
      If your change affects other repositories, such as [Documentation](https://github.com/SteeltoeOSS/Documentation), [Samples](https://github.com/SteeltoeOSS/Samples) and/or [MainSite](https://github.com/SteeltoeOSS/MainSite), add linked PRs here.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.
- [ ] You've added required license files and/or file headers (explaining where the code came from with proper attribution), where code is copied from StackOverflow, a blog, or OSS.

### CODE REVIEW

**Comment by hananiel** [comment]
I will backport to 3.x once approved.

File: src/Messaging/test/RabbitMQ.Test/Listener/DirectMessageListenerContainerIntegrationTest.cs [hunk 1]
```diff
@@ -67,6 +70,34 @@ public async Task TestDirect()
...
+        Assert.True(container.StartedLatch.Wait(TimeSpan.FromSeconds(10)));
+        var template = new RabbitTemplate(cf);
+        template.ConvertSendAndReceive<string>(Q1, "foo");
+
+        Assert.Equal(maxAttempts, listener.ExceptionCounter);
+
+        await container.StopAsync();
+        await template.StopAsync();
+        cf.Destroy();
+    }
```
Comment (TimHess) [inline src/Messaging/test/RabbitMQ.Test/Listener/DirectMessageListenerContainerIntegrationTest.cs:100]:
```suggestion
    }

```

File: src/Messaging/test/RabbitMQ.Test/Attributes/EnableRabbitIntegrationTest.cs [hunk 2]
```diff
@@ -826,13 +826,16 @@ private ServiceCollection CreateContainer(IConfiguration configuration = null)
             });
 
             // Add default container factory
+            // In Spring RetryTemplate and RecoveryCallback are setup by default
+            // However the behaviour of Retry on exception is handled by setting the advice chain (AOP) in spring, but in dotnet it will be used if present
+            // To resolve this, we are removing the retryTemplate setup here and using a different container to test relevant functionality: AutoSimpleDeclareAnonymousQueue
+            // now uses txListenerContainerFactory
+
```
Comment (TimHess) [inline src/Messaging/test/RabbitMQ.Test/Attributes/EnableRabbitIntegrationTest.cs:833]:
```suggestion
            // Add default Rabbit listener container factory.  In Spring, RetryTemplate and RecoveryCallback are setup by default.
            // In Steeltoe, no retry policy is added by default (additional configuration is required).
            // To resolve this difference, we do not setup the retryTemplate here.
            // Tests that need to work with retries now use txListenerContainerFactory

```

**Review by TimHess — APPROVED** [review]
LGTM

**Pull request merged**